sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/EnumTrailingCommaCheck.java

```diff
@@ -0,0 +1,159 @@
+ *     DELTA,; // violation
+ * }
+ * </pre>
+ * In this case the semicolon should be removed. However, if there is more in the enum body, the
+ * semicolon should be placed on a line by itself.
+ *
+ * <p>An example of how to configure the check is:
+ * <pre>
+ * &lt;module name="EnumTrailingComma"/&gt;
+ * </pre>
+ *
+ * <p>Please note that using this check together with {@code NoWhitespaceBefore} or
+ * {@code SeparatorWrap} may create conflicts with enums that contain a body:
+ * {@code EnumTrailingComma} enforces the semicolon on a separate line while
+ * {@code NoWhiteSpaceBefore} does not allow the semicolon to be preceded with whitespace and
+ * {@code SeparatorWrap} expects the semicolon to be on the same line as the last enum constant.
+ *
+ * @author <a href="kariem.hussein@gmail.com">Kariem Hussein</a>
+ */
+public class EnumTrailingCommaCheck extends AbstractCheck {
+
+    /** Key for warning message text in "messages.properties" file. */
+    public static final String MSG_KEY = "enum.trailing.comma";
+    /** Key for warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_SEMI = "enum.trailing.comma.semi";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {TokenTypes.ENUM_DEF};
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST enumDef) {
+        final DetailAST enumConstBlock = enumDef.findFirstToken(TokenTypes.OBJBLOCK);
+
+        final DetailAST enumConstLeft = enumConstBlock.findFirstToken(TokenTypes.LCURLY);
+        final DetailAST enumConstRight = enumConstBlock.findFirstToken(TokenTypes.RCURLY);
+
+        // Only check, if block is multi-line and there are more than one enum constants
+        if (enumConstLeft.getLineNo() != enumConstRight.getLineNo()
+                && enumConstBlock.getChildCount(TokenTypes.ENUM_CONSTANT_DEF) > 1) {
+            final DetailAST constant = enumConstBlock.findFirstToken(TokenTypes.ENUM_CONSTANT_DEF);
+            final DetailAST lastComma = getLastComma(constant);
+
+            final DetailAST next = lastComma.getNextSibling();
+            final int type = next.getType();
+
+            if (type == TokenTypes.SEMI) {
+                if (next.getLineNo() == lastComma.getLineNo()) {
+                    // semi on the same line as last comma
+                    log(next.getLineNo(), MSG_KEY_SEMI);
+                }
+
+            }
+            else if (type == TokenTypes.ENUM_CONSTANT_DEF) {
+                log(next.getLineNo(), MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * Get the last comma in a series of siblings.
+     *
+     * @param start the first sibling
+     * @return the AST containing the last comma
+     */
+    private DetailAST getLastComma(DetailAST start) {
```

Comment:
please do me a favor ....
make it as static method like `getLast(start, TokenType.Comma)` it might be moved eventually to some Utils class.

Reply:
I don't think this should be part of this PR.

Reply:
make it static in this PR, and moving to another Util file is another PR.

Reply:
Done: e06d5c8029a93c71f12fc0ff366a92eb683c7de2